crates/hotfix/tests/session_tests.rs

```
@@ -10,14 +10,15 @@ use hotfix_message::fix44::MSG_TYPE;
 
 mod common;
 
+const HEARTBEAT_INTERVAL: u64 = 30;
+
 #[tokio::test]
 async fn test_happy_login_flow() {
-    let session = create_session();
-    let mut mock_counterparty = MockCounterparty::start(session.clone()).await;
+    let (session, mut mock_counterparty) = setup().await;
 
     // assert that a logon message is received (type 'A')
     mock_counterparty
-        .assert_next(|msg| msg.header().get::<&str>(MSG_TYPE).unwrap() == "A")
+        .assert_next(|msg| assert_eq!(msg.header().get::<&str>(MSG_TYPE).unwrap(), "A"))
```

Moving the actual assertions into the closure for two reasons:
1. they provide much better feedback - the panic is reported as coming from this line in the test rather than the generic location inside the mock counterparty
2. allow you to use `assert_eq` with the nicer failure message
3. a single `bool` return value is awkward when you want to apply assertions to multiple fields

---

Agreed. This reads much better